source/particle/world.cc

```diff
@@ -721,21 +879,55 @@ namespace aspect
         temperature(mapping_info,
                     simulator.get_fe(),
                     simulator.introspection().component_indices.temperature),
-        compositions(mapping_info,
-                     simulator.get_fe(),
-                     simulator.n_compositional_fields() > 0 ? simulator.introspection().component_indices.compositional_fields[0] : simulator.introspection().component_indices.temperature),
 
         melt_component_indices(),
         simulator_access(simulator)
       {
-        // Create the evaluators for all compositional fields beyond the ones this class was
-        // instantiated for
+        // Create the evaluators for all compositional fields
         const unsigned int n_total_compositional_fields = simulator_access.n_compositional_fields();
+
         const auto &component_indices = simulator_access.introspection().component_indices.compositional_fields;
-        for (unsigned int composition = n_compositional_fields; composition < n_total_compositional_fields; ++composition)
-          additional_compositions.emplace_back(FEPointEvaluation<1, dim>(mapping_info,
-                                                                         simulator_access.get_fe(),
-                                                                         component_indices[composition]));
+
+        if (false)
+          {
+            // No grouping of compositional fields of same type:
+            for (unsigned int composition = 0; composition < n_total_compositional_fields; ++composition)
+              compositions.emplace_back(make<dim>(mapping_info,
+                                                  simulator_access.get_fe(),
+                                                  component_indices[composition],
+                                                  1
+                                                 ));
+          }
+        else
+          {
+            // We consider each group of consecutive compositions of the same time together. This is because it is more efficient
```

Comment:
```suggestion
            // We consider each group of consecutive compositions of the same degree together. This is because it is more efficient
```